src/hooks/useDraftSpecEditor.ts

```diff
@@ -128,12 +129,6 @@ function useDraftSpecEditor(
         }
     }, [currentCatalog, draftSpecs, entityName, setSpecs]);
 
-    useEffect(() => {
-        if (currentCatalog && !isEqual(draftSpecRef.current, currentCatalog)) {
-            setDraftSpec(currentCatalog);
-        }
-    }, [currentCatalog]);
```

Comment:
I am still testing this out. I am 99.99% sure we are good to remove this now that we are firing a mutate. Since this was kind of a janky way to get the draft spec updated anyway.